src/test/java/org/apache/sling/event/impl/jobs/JobManagerImplTest.java

```diff
@@ -19,27 +19,162 @@
+
+		jobManager.addJob("is/assigned", null, null);
+		assertFalse(logContainsLine(line -> line
+				.contains("Persisting job Sling Job [topic=is/assigned] into queue null with no assigned target")));
+	}
+
+	@Test
+	public void logsInfoForUnassigned() throws IllegalAccessException, IOException, NoSuchMethodException,
+			SecurityException, IllegalArgumentException, InvocationTargetException {
+		JobManagerImpl jobManager = getJobManager(LEVEL_INFO);
+		jobManager.addJob("not/assigned", null, null);
+		assertTrue(logContainsLine(line -> line
+				.contains("Persisting job Sling Job [topic=not/assigned] into queue null with no assigned target")));
+
+		jobManager.addJob("is/assigned", null, null);
+		assertTrue(logContainsLine(line -> !line
+				.contains("Persisting job Sling Job [topic=is/assigned] into queue null with no assigned target")));
+	}
+
+	@Test
+	public void logsDebugForAssigned() throws IllegalAccessException, IOException, NoSuchMethodException,
+			SecurityException, IllegalArgumentException, InvocationTargetException {
+		JobManagerImpl jobManager = getJobManager(LEVEL_DEBUG);
+
+		jobManager.addJob("is/assigned", null, null);
+		assertTrue(logContainsLine(line -> line
+				.contains("Persisting job Sling Job [topic=is/assigned] into queue null, target=assigned")));
+	}
+
+	@After
+	public void after() throws NoSuchMethodException, IllegalAccessException, InvocationTargetException {
+		System.setErr(originalErr);
+	}
+
+	public JobManagerImpl getJobManager(int level) throws IllegalAccessException, NoSuchMethodException,
+			SecurityException, IllegalArgumentException, InvocationTargetException {
+
+		JobManagerImpl jobManager = new JobManagerImpl();
+		
+		Object logger = FieldUtils.readDeclaredField(jobManager, "logger", true);
+		FieldUtils.writeDeclaredField(logger, "currentLogLevel", level, true);
+
+		FieldUtils.writeDeclaredField(jobManager, "configuration", configuration, true);
+		return jobManager;
+	}
 
-    private static final String QUERY_ROOT = "/var/eventing/foobar";
-    private static final QueryType QUERY_TYPE = QueryType.ACTIVE;
-    
-    // SLING-8413
-    @Test
-    public void testTopicEscaping() {
-        String baseQuery = JobManagerImpl.buildBaseQuery(QUERY_ROOT, "randomNonQuotedTopic", QUERY_TYPE, false);
-        assertEquals("/jcr:root/var/eventing/foobar/element(*,slingevent:Job)[@event.job.topic = "
-                + "'randomNonQuotedTopic' and not(@slingevent:finishedState) and @event.job.started.time",baseQuery);
-
-        String baseQuery2 = JobManagerImpl.buildBaseQuery(QUERY_ROOT, "random'Topic", QUERY_TYPE, false);
-        assertEquals("/jcr:root/var/eventing/foobar/element(*,slingevent:Job)[@event.job.topic = "
-                + "'random''Topic' and not(@slingevent:finishedState) and @event.job.started.time",baseQuery2);
-    
-    }
-    
+	private boolean logContainsLine(Predicate<String> matcher) throws IOException {
```

Comment:
This method is an exact duplicate of the same method in `JobhandlingIT.containsLine()`.

I think we can extract them to a common location to avoid code duplication.